examples/inspection/plot_permutation_importance_SHAP.py

```diff
@@ -0,0 +1,336 @@
+# possible subset sizes (n_features - 1), :math:`z'\in\{0,1\}^M`
+# (denotes the presence or absence of each feature) and :math:`\phi_i` is the
+# estimated Shapley value.
+#
+# This is much more computationally expensive than permutation importance
+# particularly as the number of possible combinations of features for all
+# feature subsets sizes quickly becomes very large as the number of features
+# increases. However, this does enable SHAP to account for interaction effects
+# between features.
+#
+# First, we will instantiate ``KernelSHAP`` using our fitted regressor,
+# ``reg`` and some background data, which will be used to simulate missingness.
+# If your dataset is small (e.g., training data is <100 samples) you can use
+# the whole training subset for the background data. Our data is larger than
+# this so we must summarize it in some way, otherwise computation will be
+# too slow. For simplicity, we will use the
+# median values of our features but SHAP offers a ``kmeans`` function that
+# can summarize each feature with ``k`` means.
+
+import shap
+
+med = X_train.median().values.reshape((1, X_train.shape[1]))
+explainer = shap.KernelExplainer(reg.predict, med)
+
+# %%
+# ``explainer`` stores various information about the data as attributes. Of
+# interest is ``expected_value``. This represents :math:`\phi_0` in our
+# equation above.
+
+explainer.expected_value
+
+# %%
+# Can you work out how this value was calculated?
+#
+# It is the prediction output by our model when using the background
+# values we gave it.
+
+reg.predict(med)
+
+# %%
+# Next we will calculate Shapley values using the testing subset. This is the
+# computationally expensive step.
+
+shap_values = explainer.shap_values(X_test, l1_reg="aic")
+
+# %%
+# Let's look at the Shapley values of one sample. There are 8 Shapley values,
+# one for each feature. '0' means that the feature did not contribute to the
+# prediction output. A negative value 'pushes' the prediction lower and a
+# positive value pushes the prediction higher.
+
+shap_values[0, :]
+
+# %%
+# The Shapley values should sum to the difference between the
+# prediction output by our our model ``reg`` and  the ``expected_value`
+# (depending on how well the linear model was able to be fit).
+
+print(f"The sum of Shapley values: {shap_values[0, :].sum()}")
+prediction = reg.predict(X_test.iloc[0, :].values.reshape(1, -1))[0]
+print(
+    "Difference between prediction and expected value: "
+    f"{prediction - explainer.expected_value}"
+)
+
+# %%
+# We can also plot the Shapley values:
+
+import matplotlib.pyplot as plt
+
+shap.summary_plot(shap_values, X_test, show=False)
+plt.tight_layout()
+
+# %%
+# In the plot above, each dot represents the Shapley value of one sample,
+# for that feature. The features are ordered from most important at the top
+# to least important at the bottom. Note that the dots cluster around 0
+# (no contribution) more and more as you go down.
```

Comment:
It might be interesting to go into details about the most important feature and interpret it in detail.